src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -148,7 +152,7 @@ public CborReaderState Peek()
                 CborMajorType.Map => CborReaderState.StartMap,
                 CborMajorType.Tag => CborReaderState.Tag,
                 CborMajorType.Special => MapSpecialValueTagToReaderState(initialByte.AdditionalInfo),
-                _ => CborReaderState.FormatError,
+                _ => throw new Exception("CborReader internal error. Invalid major type."),
```

Comment:
NotSupportedException ?

Reply:
This should be unreachable code. We plan to guard this (and a few other cases) with an assert as soon as we move the implementation to product code.